### PR TITLE
3 cost optimization runbooks

### PR DESCRIPTION
**Pull request by dougsillars**
Three AWS Runbooks around cost optimization

### CODE REVIEW

File: AWS/aws_redshift_get_daily_product_costs.ipynb [hunk 1]
```diff
@@ -0,0 +1,940 @@
+{
```
Comment (amit-chandak-unskript) [inline AWS/aws_redshift_get_daily_product_costs.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.        # Input param validation.
</pre><p>I dont understand this runbook. Doesnt it require the customer to have redshift configured to analyze AWS costs? Is there a runbook to configure AWS to do that? I am not sure how can someone use this runbook in their environment? How should be the schema for redshift n all?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/unskript/Awesome-CloudOps-Automation/pull/331/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: AWS/aws_redshift_get_ec2_daily_costs.ipynb [hunk 2]
```diff
@@ -0,0 +1,948 @@
+{
```
Comment (amit-chandak-unskript) [inline AWS/aws_redshift_get_ec2_daily_costs.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #53.        "secret_name": "\\"awsuser-doug-redshift\\""
</pre><p>this wont work, secret_name should be a parameter to this runbook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/unskript/Awesome-CloudOps-Automation/pull/331/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: AWS/aws_redshift_update_database.ipynb [hunk 3]
```diff
@@ -0,0 +1,1002 @@
+{
```
Comment (amit-chandak-unskript) [inline AWS/aws_redshift_update_database.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #28.    RebuildSql = f"copy {tableName} from 's3://unskript-billing-doug/all/unskript-billing-doug/{dateRange}/unskript-billing-doug-RedshiftManifest.json' credentials     'aws_iam_role=arn:aws:iam::100498623390:role/service-role/AmazonRedshift-CommandsAccessRole-20230103T181457' region 'us-west-2'    GZIP CSV IGNOREHEADER 1 TIMEFORMAT 'auto' manifest;"
</pre><p>Please recheck these runbooks and remove hard coding</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/unskript/Awesome-CloudOps-Automation/pull/331/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: AWS/aws_redshift_get_daily_product_costs.ipynb (hunk 1 above)
Comment (dougsillars) [reply AWS/aws_redshift_get_daily_product_costs.ipynb:1]: I can link to the blog posts that describe it all.

but yes - you do need a cost and usage report to be in Redshift for this to work

**Review by amit-chandak-unskript — APPROVED** [review]

File: AWS/aws_redshift_get_daily_product_costs.ipynb [hunk 4]
```diff
@@ -0,0 +1,1241 @@
...
+    },
+    "orderProperties": [],
+    "tags": [
+     "unSkript:nbParam"
+    ],
+    "title": "unSkript Internal"
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
```
Comment (amit-chandak-unskript) [inline AWS/aws_redshift_get_daily_product_costs.ipynb:21]: The first cell needs to be removed from the runbook. @jayasimha-raghavan-unskript please share the script to sanitize the runbooks.

**Comment by dougsillars** [comment]
@amit-chandak-unskript runbooks added without sanitization - so they should be good once the action is completed

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**